code/datums/status_effects/debuffs/fire_stacks.dm

```diff
@@ -0,0 +1,373 @@
+/datum/status_effect/fire_handler
+	duration = -1
+	alert_type = null
+	status_type = STATUS_EFFECT_REFRESH //Custom code
+	on_remove_on_mob_delete = TRUE
+	tick_interval = 2 SECONDS
+	/// Current amount of stacks we have
+	var/stacks
+	/// Maximum of stacks that we could possibly get
+	var/stack_limit = 20
+	/// What status effect types do we remove uppon being applied. These are just deleted without any deduction from our or their stacks when forced.
+	var/list/enemy_types = list()
+	/// What status effect types do we merge into if they exist. Ignored when forced.
+	var/list/merge_types = list()
+	/// What status effect types do we override if they exist. These are simply deleted when forced.
+	var/list/override_types = list()
```

Comment:
Is there any reason these need to be lists on init?
It adds a bit of extra cost to New, in general if it's not inconvienient it's best to avoid creating lists until you actually need them
for the record `for(var/thing in null)` will not runtime.